meta-mender-tegra/recipes-bsp/u-boot/u-boot-mender-tegra-vars.inc

```diff
@@ -0,0 +1,31 @@
+MENDER_UBOOT_AUTO_CONFIGURE = "0"
+# Use the mmcblk0boot1 partition for uboot environment (partition 2
+# in uboot) or user partition (0) on Nano with SDcard environment
+MENDER_UBOOT_CONFIG_SYS_MMC_ENV_PART = "${@'0' if (d.getVar('TEGRA_MENDER_UBOOT_ENV_IN_SPIFLASH') or '') != '1' and (d.getVar('TEGRA_SPIFLASH_BOOT') or '') == '1' else '2'}"
+
+# Calculate this offset by adding up the offsets of each partition preceeding the uboot_env partition in sdmmc_boot and aligning to the next
+# 4096 byte boundary, then subtracting 4 MiB (4194304) since the sdmmc_boot represents the combined boot0 and boot1 partitions
+# Please note the suggestions in the nvidia thread at https://devtalk.nvidia.com/default/topic/1063652/jetson-tx2/mmcblk0boot1-usage-at-address-4177408-and-u-boot-parameter-storage-space-availability/
+# regarding suggestions about locating this environment area in mmcblk0boot1 at 0x3BB000 (3911680)
+# We use the same offset for tegra210 eMMC. For tegra210 SPI/SD, we
+# locate the environment in the padding block, before the VER
+# partition when using SPI flash.  Otherwise, we use the location of
+# the ENV partition on the SDcard - 0xC00000 (12582912).
```

Comment:
Since uboot/TX2 will be the only platforms that support u-boot and are detailed above I think there won't be any scenarios where we locate at 0xC00000.  Is that correct?

Reply:
t210 (tx1/nano) also uses u-boot, and that's the offset used there when the conditions are met for env on sdcard https://github.com/OE4T/meta-mender-community/blob/15635b9d9ed8d9064d32a5e3763b355520e54359/meta-mender-tegra/recipes-bsp/u-boot/u-boot-mender-tegra-vars.inc#L15

Reply:
Sorry I meant nano/TX2.  Apparently I did this code revew before I finished waking up.

So the way we use 0xC00000 would be if `TEGRA_MENDER_UBOOT_ENV_IN_SPIFLASH` is 0 and `TEGRA_SPIFLASH_BOOT` is 1, which would mean [jetson-nano-qspi-sd](https://github.com/madisongh/meta-tegra/blob/33bdeaf28676a5cdad49733a2d29faecf8b3f350/conf/machine/jetson-nano-qspi-sd.conf#L35) with override as described [here](https://github.com/OE4T/meta-mender-community/blob/master%2Btegra-updates/meta-mender-tegra/recipes-bsp/u-boot/u-boot-mender-tegra-vars.inc#L27) in the build configuration.

This would mean partition layout defined [here](https://github.com/mendersoftware/meta-mender-community/blob/496604fd0d5683a247e6c356605d3ad32faf220e/meta-mender-tegra/recipes-bsp/tegra-binaries/mender-custom-flash-layout/jetson-nano-qspi-sd/flash_mender.xml#L88).  Wouldn't offset 12582912 put us somewhere in the APP partition?  Or am I still confusing myself?  I realize this didn't change from the previous release.

Reply:
The Nano SDcard layout puts the APP partition at the end of the card, rather than at the beginning (taking advantage of GPT's ability to have partition numbering in the table be independent of the physical ordering of the partitions on disk).

Reply:
I see the stock layout at `tmp/work-shared/L4T-tegra210-32.4.3-r0/Linux_for_Tegra/bootloader/t210ref/cfg/flash_l4t_t210_max-spi_sd_p3448.xml` has this content for the sdcard:
```
    <device type="sdcard" instance="0">
        <partition name="GP1" id="20" type="GP1">
            <allocation_policy> sequential </allocation_policy>
            <filesystem_type> basic </filesystem_type>
            <size> 2097152 </size>
            <file_system_attribute> 0 </file_system_attribute>
            <allocation_attribute> 8 </allocation_attribute>
            <percent_reserved> 0 </percent_reserved>
        </partition>

        <partition name="APP" id="21" type="data">
            <allocation_policy> sequential </allocation_policy>
            <filesystem_type> basic </filesystem_type>
            <size> APPSIZE </size>
            <file_system_attribute> 0 </file_system_attribute>
            <allocation_attribute> 0x8 </allocation_attribute>
            <percent_reserved> 0 </percent_reserved>
            <align_boundary> 4096 </align_boundary>
            <unique_guid> APPUUID </unique_guid>
            <filename> APPFILE </filename>
        </partition>

        <partition name="GPT" id="22" type="GPT">
            <allocation_policy> sequential </allocation_policy>
            <filesystem_type> basic </filesystem_type>
            <size> 2097152 </size>
            <file_system_attribute> 0 </file_system_attribute>
            <allocation_attribute> 8 </allocation_attribute>
            <percent_reserved> 0 </percent_reserved>
        </partition>
    </device>
```
Which essentially matches the layout in https://github.com/mendersoftware/meta-mender-community/blob/496604fd0d5683a247e6c356605d3ad32faf220e/meta-mender-tegra/recipes-bsp/tegra-binaries/mender-custom-flash-layout/jetson-nano-qspi-sd/flash_mender.xml#L101 for the first two partitions (GP1 and APP)

Reply:
Resolving this as we have a more current conversation on the same below.